llmfoundry/models/layers/ffn.py

```diff
@@ -139,24 +140,27 @@ def __init__(
             expansion_ratio,
             ffn_hidden_size,
         )
-        self.fc_kwargs: dict[str, Any] = {
-            'bias': bias,
-        }
 
-        self.fc_kwargs['device'] = device
+        if fc_type is None:
+            self.fc_type = copy.deepcopy(fc_type_defaults)
+            self.fc_type['bias'] = bias
+            self.fc_type['device'] = device
+        else:
+            self.fc_type = fc_type
+        self.fc_type_name = self.fc_type['name']
```

Comment:
any reason you don't pop the name off here so you can avoid the filter in the builder?

Reply:
If we pop off the name from the dict in layer 1's FFN, then layer 2's FFN will not have the `name` key since it's been popped off already and the dict is shared between the two. So if I want to pop the name off, I'd have to `copy.deepcopy` the `fc_type` dict every time in both attention and ffn. Unless there's a different way?

Reply:
yeah i guess the way you have it is fine